packages/workspace/src/generators/new/new.ts

```diff
@@ -302,11 +311,19 @@ function addPresetDependencies(host: Tree, options: Schema) {
   );
 }
 
-function normalizeOptions(options: Schema): Schema {
+function normalizeOptions(options: NormalizedSchema): NormalizedSchema {
   options.name = names(options.name).fileName;
   if (!options.directory) {
     options.directory = options.name;
   }
+  const { preset } = options;
+  if (preset.match(/.+@/)) {
+    // If the preset already contains a version in the name
+    // -- my-package@2.0.1
+    // -- @scope/package@version
+    options.preset = preset[0] + preset.substring(1).split('@')[0];
+    options.presetVersion = preset.split('@')[1];
```

Comment:
I think this would match  `acme/pkg` for  `'@acme/pkg@1.2.3' instead of `'1.2.3'`?

Reply:
Like this?

```
const match = preset.match(
  /^(?<package>(@.+\/)?[^@]+)(@(?<version>\d+\.\d+\.\d+))?$/
);

if (!match) {
  return preset;
}

const { package, version } = match;
return version ? { preset: package, presetVersion: version } : { preset };
```

Reply:
No, it shouldn't since I use the `preset[0]` to capture the `@`  i.e. the first character
And `substring(1).split('@')[0]` would give `acme/pkg`